lib/compat/wordpress-5.9/class-gutenberg-rest-navigation-controller.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Templates_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Templates REST API Controller.
+ */
+class Gutenberg_REST_Navigation_Controller extends WP_REST_Posts_Controller {
```

Comment:
This is quick&dirty attempt to reproduce template areas mechanics of requesting items by their slug, e.g. `/wp/v2/navigation/header-menu`. A lot of cleaning up needed here.